analysis/model/src/test/kotlin/de/maibornwolff/codecharta/serialization/ProjectSerializerTest.kt

```diff
@@ -15,47 +25,108 @@ import kotlin.io.path.absolute
 import kotlin.io.path.createTempDirectory
 import kotlin.test.assertTrue
 
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
 class ProjectSerializerTest {
-    val EXAMPLE_JSON_VERSION_1_3 = "example_api_version_1.3.cc.json"
-    val tempDir = createTempDirectory()
-    val filename = tempDir.absolute().toString() + "test.cc.json"
-    val project = mockk<Project>()
+    private val EXAMPLE_JSON_VERSION_1_3 = "example_api_version_1.3.cc.json"
+    private val tempDir = createTempDirectory()
+    private val filename = tempDir.absolute().toString() + "test.cc.json"
+    private val project = mockk<Project>()
+    private val loggerMock = mockk<KLogger>()
+    private val infoMessagesLogged = mutableListOf<String>()
+
+    @BeforeAll
```

Comment:
- The object under test is ProjectSerializer, which is declared as an object
- According to the documentation, object initialization happens upon first access
- Since all tests use ProjectSerializer, it's unclear whether a test initializes it or uses an already initialized instance
- To make sure we use a mocked logger for testing the logging behaviour in the last test, we need to setup the logger mock before the initialization of the ProjectSerializer (aka before any test runs)

I'd like to know what you think about how we're testing things and whether you think we should add a comment to explain the code